src/lib.rs

```diff
@@ -62,6 +63,10 @@ pub fn set_global_codegen_config(codegen_config: &CodegenConfig) {
     }
 }
 
+fn is_valid_pass(pass: &String) -> bool {
+    unsafe { ffi::BinaryenShimPassFound(CString::new(pass.as_str()).unwrap().as_ptr()) }
```

Comment:
This might be not sound because by the time `BinaryenShimPassFound` is called, the `CString` is already deallocated, and thus the pointer returned by `as_ptr` is invalidated. I think this might be solved by this code:

```suggestion
    let c_str = CString::new(pass.as_str()).unwrap();
    unsafe { ffi::BinaryenShimPassFound(c_str.as_ptr()) }
```

Reply:
Funny I just did this a second ago :)

Reply:
Will rename it to `cstr`.